antarest/study/business/link_management.py

```diff
@@ -76,6 +78,49 @@ def create_link(self, study: Study, link_creation_dto: LinkDTO) -> LinkDTO:
 
         return link_creation_dto
 
+    def update_link(self, study: RawStudy, link_dto: LinkDTO) -> LinkDTO:
+        link = link_dto.to_internal(StudyVersion.parse(study.version))
+        file_study = self.storage_service.get_storage(study).get_raw(study)
+
+        self.check_link_existence(file_study, link)
+
+        command = UpdateLink(
+            area1=link.area1,
+            area2=link.area2,
+            parameters=link.model_dump(
+                include=link_dto.model_fields_set, exclude={"area1", "area2"}, exclude_none=True
+            ),
+            command_context=self.storage_service.variant_study_service.command_factory.command_context,
+            study_version=file_study.config.version,
+        )
+
+        execute_or_add_commands(study, file_study, [command], self.storage_service)
+
+        updated_link = self.get_internal_link(study, link)
+
+        return updated_link.to_dto()
+
+    def check_link_existence(self, file_study: FileStudy, link: LinkInternal) -> None:
+        area_from, area_to = sorted([link.area1, link.area2])
```

Comment:
You shouldn't have to sort this anymore as you did it inside the Area pydantic class

Same goes for other parts of the code I'll let you check